### ISSUE TITLE
Unexpected "Error: Division by zero"

### ISSUE THREAD

**Issue by pmor13** [cc65/cc65#1768]
This code:
```
#if 1 || (8 / 0)
#endif
```
leads to:
```
Error: Division by zero
```
while per C11 the `(8 / 0)` is not evaluated (emphasis added):
> If the first operand compares unequal to 0, the **second operand is not evaluated**.

Hence, the `Error: Division by zero` may be unexpected.

Note: the same for `#if 0 && (8 / 0)`, `1 ? 0 : 1 / 0`, and so on.

**Comment by acqn** [comment]
Not only in PP, but also:
```c
int a = 1 || (8 / 0);
int b = 0 && (8 % 0);
int c = 1 ? 42 : (0 % 0);
int d = 1 || a / 0;
int e = 0 && b % 0;
int f = 1 ? 42 : (a %= 0, b /= 0);
```
since the PP is reusing "almost the same" expression parser.
Somewhat clean fix would be best with separation of the parser and the codegen.

**Comment by mrdudz** [comment]
fixed by #1913
